### ISSUE TITLE
Pub times out in China

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#780]
_Originally opened as dart-lang/sdk#15638_

*This issue was originally filed by bitni...&#064;gmail.com*
_____
**What steps will reproduce the problem?**
1.open dart editor 1.0.0_r30798
2.add chrome_gen to pubspec.yaml
3.click Run Pub get

**What is the expected output? What do you see instead?**

--- 2013-12-14 20:17:03 Running pub get ... ---
Pub get failed, [69] Resolving dependencies...............
Downloading chrome_gen 0.3.1 from hosted...
Downloading browser 0.9.0 from hosted...
Error in PubHttpClient.send (issue dart-lang/sdk#12581) error: SocketException: OS Error: 信号灯超时时间已到(those words means signal light time out)

, errno = 121, address = commondatastorage.googleapis.com, port = 5861
&nbsp;&nbsp;&nbsp;&nbsp;stacktrace: #­0      PubHttpClient.send (file:///e:/b/build/slave/dart-editor-win-stable/build/dart/sdk/lib/_internal/pub/lib/src/http.dart:60)
#­1      HostedSource.get.&lt;anonymous closure&gt; (file:///e:/b/build/slave/dart-editor-win-stable/build/dart/sdk/lib/_internal/pub/lib/src/source/hosted.dart:93)
#­2      Future.Future.sync (dart:async/future.dart:153)
#­3      HostedSource.get (file:///e:/b/build/slave/dart-editor-win-stable/build/dart/sdk/lib/_internal/pub/lib/src/source/hosted.dart:84)
#­4      Source.downloadToSystemCache.&lt;anonymous closure&gt; (file:///e:/b/build/slave/dart-editor-win-stable/build/dart/sdk/lib/_internal/pub/lib/src/source.dart:163)
#­5      \_Future._propagateToListeners.&lt;anonymous closure&gt; (dart:async/future_impl.dart:453)
#­6      \_rootRun (dart:async/zone.dart:683)
#­7      \_RootZone.run (dart:async/zone.dart:823)
#­8      \_Future._propagateToListeners (dart:async/future_impl.dart:445)
#­9      \_Future._complete (dart:async/future_impl.dart:303)
#­10     \_Future._asyncComplete.&lt;anonymous closure&gt; (dart:async/future_impl.dart:354)
#­11     \_asyncRunCallback (dart:async/schedule_microtask.dart:18)
#­12     \_createTimer.&lt;anonymous closure&gt; (dart:async-patch/timer_patch.dart:11)
#­13     \_Timer._createTimerHandler._handleTimeout (timer_impl.dart:151)
#­14     \_Timer._createTimerHandler.&lt;anonymous closure&gt; (timer_impl.dart:166)
#­15     \_RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:93)

**What version of the product are you using? On what operating system?**
Window 7 SP1 x64

**Please provide any additional information below.**
by the way,I'm in china.

**Comment by DartBot** [comment]
<a href="https://github.com/floitschG"><img src="https://avatars.githubusercontent.com/u/8631949?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [floitschG](https://github.com/floitschG)**

----

Updated title.
______
*Added [Area-Pub](../labels/Area-Pub), [Triaged](../labels/Triaged) labels.*
*Changed the title to: "Pub times out in China".*

**Comment by DartBot** [comment]
<a href="https://github.com/floitschG"><img src="https://avatars.githubusercontent.com/u/8631949?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [floitschG](https://github.com/floitschG)**

----

Issue dart-lang/sdk#15639 has been merged into this issue.

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

Do you get this error every time you try to install, or is it intermittent?

If you go to the "versions" tab on http://pub.dartlang.org/packages/chrome_gen, are you able to download any of the archives from there?
______
*Added [NeedsInfo](../labels/NeedsInfo) label.*

**Comment by DartBot** [comment]
*This comment was originally written by bitnick...&#064;gmail.com*
_____
Yes,I got the error every time.
I can not download any of the archives from http://pub.dartlang.org/packages/chrome_gen
I also got the timeout message when I ping commondatastorage.googleapis.com(The archives's url is https://commondatastorage.googleapis.com/pub.dartlang.org/packages/chrome_gen-0.3.2.tar.gz)
I think it would be blocked by Great Firewall of China

**Comment by DartBot** [comment]
<a href="https://github.com/munificent"><img src="https://avatars.githubusercontent.com/u/46275?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [munificent](https://github.com/munificent)**

----

I don't think there's much we can do to resolve this, sorry.
______
*Added [NotPlanned](../labels/NotPlanned) label.*
